longevity_operator_multi_tenant_test.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+
+import copy
+
+from longevity_test import LongevityTest
+
+from sdcm.utils.common import ParallelObject
+
+
+# pylint: disable=too-many-instance-attributes
+class ScyllaClusterStats(LongevityTest):
+    # pylint: disable=too-many-arguments
+    def __init__(self, db_cluster, loaders, monitors, prometheus_db, params, test_config, cluster_index):
+        super().__init__(start_events_device=False)
```

Comment:
I don't think we need the super here, anything we need from the the test constructor should be supplied here,
what's globally use, is part of ` LongevityOperatorMulitiTenantTest`, i.e. the actual/origin test.

not that is solves any of the other issues we encountered

Reply:
here a unit test that help me tryout this code structure, it's not fully working flow, but it got me to the problem with missing `_testMethod`, maybe with a bit more work it would also help figure the nemesis issues we were having:

```diff
diff --git a/sdcm/db_stats.py b/sdcm/db_stats.py
index 951d2f16..7032aa45 100644
--- a/sdcm/db_stats.py
+++ b/sdcm/db_stats.py
@@ -196,9 +196,12 @@ class PrometheusDBStats():
         self.host = host
         self.port = port
         self.range_query_url = "http://{}:{}/api/v1/query_range?query=".format(normalize_ipv6_url(host), port)
-        self.config = self.get_configuration()
         self.alternator = alternator
 
+    @cached_property
+    def config(self):
+        return self.get_configuration()
+
     @property
     def scylla_scrape_interval(self):
         return int(self.config["scrape_configs"]["scylla"]["scrape_interval"][:-1])
diff --git a/unit_tests/test_tester.py b/unit_tests/test_tester.py
index 6c736a33..0ac9fe2a 100644
--- a/unit_tests/test_tester.py
+++ b/unit_tests/test_tester.py
@@ -318,3 +318,37 @@ class SubtestsSuccessTest(ClusterTesterForTests):
         super()._validate_results()
         assert self.event_summary == {'NORMAL': 2}
         assert self.final_event.test_status == 'SUCCESS'
+
+
+from longevity_operator_multi_tenant_test import LongevityOperatorMulitiTenantTest
+
+
+class FakeMultiTenantSCTConfiguration(SCTConfiguration):
+    def _load_environment_variables(self):
+        return {
+            'config_files': ['test-cases/scylla-operator/longevity-scylla-operator-3h-multitenant.yaml'],
+            'cluster_backend': 'k8s-eks',
+            'logs_transport': 'ssh'
+        }
+
+
+class FakeDbCluster(MagicMock):
+    nodes = [MagicMock(), MagicMock(), MagicMock()]
+
+    tenants_number = 2
+
+    def get_db_auth(self):
+        return (None, None)
+
+
+class TestMulitTenant(LongevityOperatorMulitiTenantTest, ClusterTesterForTests):
+
+    def init_resources(self, loader_info=None, db_info=None,
+                       monitor_info=None):
+        self.db_clusters_multitenant = [FakeDbCluster(), FakeDbCluster()]
+        self.loaders_multitenant = [FakeDbCluster(), FakeDbCluster()]
+        self.monitors_multitenant = [FakeDbCluster(), FakeDbCluster()]
+
+    def _init_params(self):
+        self.log = logging.getLogger(self.__class__.__name__)
+        self.params = FakeMultiTenantSCTConfiguration()
```

Reply:
removed `super().init`